l3kernel/l3keys.dtx

```diff
@@ -1843,22 +1844,24 @@
     \@@_property_find_err:w
   {
     \cs_set_nopar:Npe \l_keys_path_str
-      { \str_if_empty:NF \l_@@_module_str { \l_@@_module_str / } #1 }
-    \@@_property_find_auxi:w #2 \s_@@_nil \@@_property_find_auxiii:w . \s_@@_nil
-      \@@_property_find_auxiv:w
+      {
+        \str_if_empty:NF \l_@@_module_str { \l_@@_module_str / }
+        \exp_after:wN \@@_trim_spaces:n \tex_expanded:D {{
+        #1
+        \if_false: }}} \fi:
+        \@@_property_find_auxi:w #2 \s_@@_nil \@@_property_find_auxiii:w
+          . \s_@@_nil \@@_property_find_auxiv:w
   }
-\cs_new_protected:Npn \@@_property_find_auxiii:w #1 \s_@@_mark
+\cs_new:Npn \@@_property_find_auxiii:w #1 \s_@@_mark #2 . #3 \s_@@_nil #4
   {
-    \cs_set_nopar:Npe \l_keys_path_str { \l_keys_path_str . #1 }
-    \@@_property_find_auxi:w
+    . #1 #4 #2 \s_@@_mark #3 \s_@@_nil #4
   }
-\cs_new_protected:Npn \@@_property_find_auxiv:w
+\cs_new:Npn \@@_property_find_auxiv:w
     #1 \s_@@_nil \@@_property_find_auxiii:w
     \s_@@_mark \s_@@_nil \@@_property_find_auxiv:w
   {
+    \if_false: {{{ \fi: }}}
     \cs_set_nopar:Npe \l_@@_property_str { . #1 }
-    \cs_set_nopar:Npe \l_keys_path_str
-      { \exp_after:wN \@@_trim_spaces:n \exp_after:wN { \l_keys_path_str } }
     \tl_set_eq:NN \l_keys_path_tl \l_keys_path_str
   }
```

Comment:
`\@@_property_find_auxiv:w` still contains assignments, hence is not expandable.

I'll open a PR to fix this and add `\begin{macro}[EXP]` for now expandable `\@@_property_find_auxi:w`.

Reply:
Ah, it seems by using brace trick, `\@@_property_find_auxiv:w` has to be expandable.

Reply:
Correct, and it's only ever called inside of a context where the first step was already `protected`, so nothing bad really. You could (in theory) define it `protected` and then trick the system to expand it anyways (e.g., `\exp_after:wN \prg_do_nothing: \@@_property_find_auxiv:w`) but I don't see any gain there (it would however slow this solution down...).

Reply:
I agree that unnecessary protection that is then worked around for something that is only used as an internal function inside a protected function (and never exprected to be used elsewhere) is in my opinion the wrong approach. If you want, you should explicitly say that "it is not protected because..." but  the highest goal for such internal stuff is really to be fast